transcom/mymove
# INT B-20447 Customer Search Sorting Fixes

## [Agility ticket](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-20447)

## Summary

This branch fixes two problems that existed on the "Customer Search" tab (as well as other places where customer search was allowed):

1. Names weren't sorted by closest match, i.e. a search for "Ted" would have a user with the last name, first name of "Submitted, HHG" appear at the top, several spots above a user with the name "Marine, Ted". Exact matches now appear at the top of the search results.
2. Sorting on the customer name field was a combination of last name and first name, which led to changing the sort from ascending to descending not actually doing much. For example if you were to have this list in an descending sort:

- Submitted, HHG
- Submitted, PPM
- Unsubmitted, PPM

Instead of "Unsubmitted" going to the top like you might expect when you change to ascending sort, it would instead reverse the sort inside of the last name groups like this:

- Submitted, PPM
- Submitted, HHG
- Unsubmitted, PPM

Now, first name has been removed from the sort consideration, and the expected sort result occurs.

### How to test

1. Login as Service Counselor and access the "Customer Search" tab.
2. After having ran the "db_dev_e2e_populate" script if you haven't already, search for "Ted", you should see "Marine, Ted" at the very top:
![ted_search](https://github.com/user-attachments/assets/bf942506-c4a5-4b81-8b7f-eb30d08321de)

3. Clicking the customer name sort column once should give you the following result: 
![ted_search_first_sort](https://github.com/user-attachments/assets/f86318b9-593c-4c9d-9f77-e46720fec30b)

4. Clicking on the column again should now give you this:
![ted_search_second_sort](https://github.com/user-attachments/assets/75f79af5-12a8-4e4c-a474-eca673463463)

5. You can try this with any other names for customers that are present and ensure you get the expected result. All other columns should sort as expected as well.

6. If no results are returned, there should be a message in the table indicating this.
